runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/LyftProcessJobBundleFactory.java

```diff
@@ -0,0 +1,147 @@
+    try {
+      JOB_INFO.set(jobInfo);
+      return new LyftProcessJobBundleFactory(jobInfo);
+    } finally {
+      JOB_INFO.remove();
+    }
+  }
+
+  private LyftProcessJobBundleFactory(JobInfo jobInfo) throws Exception {
+    super(jobInfo);
+  }
+
+  @Override
+  protected EnvironmentFactory getEnvironmentFactory(
+      GrpcFnServer<FnApiControlClientPoolService> controlServiceServer,
+      GrpcFnServer<GrpcLoggingService> loggingServiceServer,
+      GrpcFnServer<ArtifactRetrievalService> retrievalServiceServer,
+      GrpcFnServer<StaticGrpcProvisionService> provisioningServiceServer,
+      ControlClientPool.Source clientSource,
+      IdGenerator idGenerator) {
+
+    return new LyftPythonEnvironmentFactory(
+        Preconditions.checkNotNull(JOB_INFO.get(), "jobInfo is null"),
+        controlServiceServer,
+        loggingServiceServer,
+        retrievalServiceServer,
+        provisioningServiceServer,
+        idGenerator,
+        clientSource);
+  }
+
+  private static class LyftPythonEnvironmentFactory implements EnvironmentFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(LyftPythonEnvironmentFactory.class);
+
+    private final JobInfo jobInfo;
+    private final ProcessManager processManager;
+    private final GrpcFnServer<FnApiControlClientPoolService> controlServiceServer;
+    private final GrpcFnServer<GrpcLoggingService> loggingServiceServer;
+    private final GrpcFnServer<ArtifactRetrievalService> retrievalServiceServer;
+    private final GrpcFnServer<StaticGrpcProvisionService> provisioningServiceServer;
+    private final IdGenerator idGenerator;
+    private final ControlClientPool.Source clientSource;
+
+    private LyftPythonEnvironmentFactory(
+        JobInfo jobInfo,
+        GrpcFnServer<FnApiControlClientPoolService> controlServiceServer,
+        GrpcFnServer<GrpcLoggingService> loggingServiceServer,
+        GrpcFnServer<ArtifactRetrievalService> retrievalServiceServer,
+        GrpcFnServer<StaticGrpcProvisionService> provisioningServiceServer,
+        IdGenerator idGenerator,
+        ControlClientPool.Source clientSource) {
+      this.jobInfo = jobInfo;
+      this.processManager = ProcessManager.create();
+      this.controlServiceServer = controlServiceServer;
+      this.loggingServiceServer = loggingServiceServer;
+      this.retrievalServiceServer = retrievalServiceServer;
+      this.provisioningServiceServer = provisioningServiceServer;
+      this.idGenerator = idGenerator;
+      this.clientSource = clientSource;
+    }
+
+    /** Creates a new, active {@link RemoteEnvironment} backed by a forked process. */
+    @Override
+    public RemoteEnvironment createEnvironment(RunnerApi.Environment environment) throws Exception {
+      String workerId = idGenerator.getId();
+
+      String pipelineOptionsJson = JsonFormat.printer().print(jobInfo.pipelineOptions());
+      HashMap<String, String> env = new HashMap<>();
+      env.put("WORKER_ID", workerId);
```

Comment:
I am assuming that these are well known constant and it is used other part of the SDK as well. Can we define some constants and use it here?

Reply:
They are defined in the Python SDK, not in Java.